nixos/modules/system/boot/loader/systemd-boot/systemd-boot.nix

```diff
@@ -148,15 +232,61 @@ in {
     assertions = [
       {
         assertion = (config.boot.kernelPackages.kernel.features or { efiBootStub = true; }) ? efiBootStub;
-
         message = "This kernel does not support the EFI boot stub";
       }
+      {
+        assertion = builtins.all (filename: !(hasInfix "/" filename)) (builtins.attrNames cfg.extraEntries);
+        message = "systemd-boot does not support entries within folders";
```

Comment:
if possible assertions should give an indication where the error occured to make it easier to find. all of these would have to be split to generate one assertion per condition and thing checked to allow for that, eg with error messages like
```
boot.loader.systemd-boot.extraEntries.${lib.strings.escapeNixIdentifier attr} is invalid: entries within folders are not supported
```

Reply:
Fixed :+1: